Game/src/Enemy.java

```diff
@@ -0,0 +1,172 @@
+
+    // Speed
+    private double speed = 2.0;
+
+    // Velocities
+    private double vX, vY;
+
+    // Health Points
+    private int hp = 100;
+
+    // Enemy is dead
+    private boolean isDead = false;
+
+    // HP Display
+    public Rectangle hpBar = new Rectangle();
+
+    // Enemy Logic
+    private LogicMode logicMode;
+
+    // Logic Options
+    private double pointA, pointB, oPointA, oPointB;
+    private boolean flag;
+    private double patrolDistance;
+    private boolean patrolTriggered = false;
+
+    Enemy(double x, double y, double width, double height, Image sprite, LogicMode logicMode) {
+        this.logicMode = logicMode;
+
+        this.setX(x);
+        this.setY(y);
+        this.setFitWidth(width);
+        this.setFitHeight(height);
+        this.setImage(sprite);
+
+        hpBar.setWidth(width);
+        hpBar.setHeight(4);
+        hpBar.setFill(Color.GREEN);
+    }
+
+    Enemy(double x, double y, Image sprite, LogicMode logicMode) {
+        this(x, y, MapLoader.GRID_SIZE, MapLoader.GRID_SIZE, sprite, logicMode);
+    }
+
+    Enemy(double x, double y, Image sprite) {
+        this(x, y, MapLoader.GRID_SIZE, MapLoader.GRID_SIZE, sprite, LogicMode.NONE);
+    }
+
+    public void onUpdate() {
+
+        vX = 0.0;
+        vY = 0.0;
+
+        pointA = oPointA - MainApplication.vStart;
+        pointB = oPointB - MainApplication.vStart;
+
+        if (logicMode == LogicMode.POINT_AB) {
+            if (getX() >= pointB) {
+                flag = false;
+            } else if (getX() <= pointA) {
+                flag = true;
+            }
+            vX += flag ? speed : -speed;
+
+        } else if (logicMode == LogicMode.FOLLOW) {
+            if (getX() < MainApplication.player.getX()) {
+                vX += speed;
+            } else if (getX() > MainApplication.player.getX()) {
+                vX -= speed;
+            }
+        } else if (logicMode == LogicMode.PATROL_AB){
+
+            if (getX() + patrolDistance > MainApplication.player.getX() && getX() - patrolDistance < MainApplication.player.getX()) {
+                patrolTriggered = true;
+            }
+
+            if (!patrolTriggered) {
+                if (getX() >= pointB) {
+                    flag = false;
+                } else if (getX() <= pointA) {
+                    flag = true;
+                }
+                vX += flag ? speed : -speed;
+            } else {
+                if (getX() < MainApplication.player.getX()) {
+                    vX += speed;
+                } else if (getX() > MainApplication.player.getX()) {
+                    vX -= speed;
+                }
+            }
+
+        }
+
+        setX(getX() + vX);
+        setY(getY() + vY);
+
+        hpBar.setX(getX());
+        hpBar.setY(getY() - 8);
+
+    }
+
+    public boolean checkCollision(Player player){
+
+        boolean collided = false;
+
+        if (player.getX() + player.getFitWidth() >= getX() && player.getX() < getX() + getWidth()
```

Comment:
Use if, else if here. Otherwise the program will have to check all of these even if the first one is true